engine/table/src/main/java/io/deephaven/engine/table/impl/hierarchical/HierarchicalTableImpl.java

```diff
@@ -848,8 +853,24 @@ private VisitAction getAction() {
             return action;
         }
 
-        private LinkedDirective addChild(@NotNull final LinkedDirective childDirective) {
-            (children == null ? children = new ArrayList<>() : children).add(childDirective);
+        private LinkedDirective addOrReplaceChild(
+                @NotNull final LinkedDirective childDirective,
+                @Nullable final LinkedDirective removedChildDirective) {
+            if (children == null) {
+                Assert.eqNull(removedChildDirective, "removedChildDirective");
+                children = new ArrayList<>();
+            }
+            if (removedChildDirective == null) {
+                childDirective.slotInParent = children.size();
+                children.add(childDirective);
+            } else {
+                Assert.eq(children.get(removedChildDirective.slotInParent),
+                        "children.get(removedChildDirective.slotInParent)",
+                        removedChildDirective, "removedChildDirective");
+                Assert.eq(removedChildDirective.action, "removedChildDirective.action", Linkage, "Linkage");
```

Comment:
i think technically you don't need the "Linkage" string, since the tostring of the enum is its own name